### PR TITLE
Matt branch

### PR DESCRIPTION
**Pull request by mattdias**

### CODE REVIEW

File: app/views/directives/home-articles-small.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+                        $scope.error    = null;
+
+                        $scope.articles = [];
+
+                        loadArticles();
+                        //---------------------------------------------------------------------
+                        function loadArticles(){
+                            var ag = [];
+                            ag.push({"$match":{"$and":[{"adminTags.title.en":encodeURIComponent($scope.tags||"ABSCH-Announcement")}]}});
+                            ag.push({"$project" : {"title":1, "content":1, "coverImage":1, "meta":1, "summary":1}});
```
Comment (blaisefonseca) [inline app/views/directives/home-articles-small.js]: ag.push({"$limit" : 10}); for limiting number of records

File: app/views/directives/home-articles-small.html [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+.noimage{display: table-cell; vertical-align: middle; font-size:24px;padding:10px; top:-20px;position: relative; }
+
+</style>
+    <h4 style="padding-bottom:5px;margin-bottom:0;padding-top:20px;">
+        Announcements
+        <span class="pull-right more-link" style="top:10px; position: relative;"> <a href="/articles/" class="color-litegrey"><i class="fa fa-chevron-circle-right" aria-hidden="true"></i> {{articles.length}} records </a></span>
+    </h4>
+    <div style="border-top:1px solid #DDD; margin-bottom:8px;">
+        <div class="row" style="max-height:180px;overflow:hidden;">
+           <div  ng-repeat="article in articles | orderBy:'meta.createdOn':true | limitTo:9" style="cursor:pointer;" class="col-xs-12 col-md-6 col-lg-4"  >
```
Comment (blaisefonseca) [inline app/views/directives/home-articles-small.html]: limit number of records in query

Comment (blaisefonseca) [reply app/views/directives/home-articles-small.html]: add sorting to query as well

File: app/views/directives/home-articles-small.js (hunk 1 above)
Comment (blaisefonseca) [reply app/views/directives/home-articles-small.js]: ag.push({ $sort   : {"meta.modifiedOn":-1})

File: app/views/directives/task-id-directive.html [hunk 3]
```diff
@@ -57,24 +58,26 @@ <h3 class="color-black;" style="margin-top:0;border-bottom:1px solid #999;">
...
+                        <div style="display:inline;padding:5px;"  ng-show="activity.name == 'deleteRecord')">
                                 <button type="button" class="btn btn-danger bold" ng-click="confirmDelete()" ng-disabled="isUpdating">
                                 <i class="fa fa-check-circle"></i>
                                 <span >Delete</span>
                             </button>
                         </div>
                     </div>
-                    <div ng-show="workFlowIsCreatedByMe()">
+
+                    <div ng-show="workFlowIsCreatedByMe() || isAdmin()">
```
Comment (blaisefonseca) [inline app/views/directives/task-id-directive.html]: what is the purpose of displaying this to Admin, they will not be able to do anything anyways?

File: app/views/forms/view/view-articles.html [hunk 4]
```diff
@@ -44,25 +44,27 @@ <h3 style="font-size:160%;margin-bottom:30px;">{{article.title  | lstring:locale
...
-                                        <div style="margin:5px;" ng-class="{'noimagetable':!article.coverImage.url}" >
-                                            
-                                            <h4  ng-class="{'noimage':!article.coverImage.url}">
-                                                <span style="display:block;font-size: 10px; text-transform: uppercase;" class="color-blue bold">{{article.summary | lstring:locale}}</span>
-                                                {{article.title | lstring:locale}}</h4>
-                                        </div>
-                                    </div>
-                    </div>  
+        <div class="row"  ng-if="articles">
+            <div  ng-repeat="article in articles | orderBy:'meta.createdOn':true | limitTo:9" style="cursor:pointer;" class="col-xs-12 col-md-6 col-lg-4"  >
```
Comment (blaisefonseca) [inline app/views/forms/view/view-articles.html]: limit records in query

File: app/views/register/dashboard.html [hunk 5]
```diff
@@ -167,180 +182,194 @@ <h4 style="padding-bottom:3px;margin-top:0;border-bottom:1px solid #ddd;">
...
-                        </tbody>
-                        </table>
-                        <div ng-if="topRecords[schema]" style="width:100%">
-                            <top-records schema="{{schema}}" top="10" style="padding-top:5px;" show-schema-name="false"></top-records>
-                        </div>
+                                    </td>
+                                    <td style="min-width:180px!important;">
+                                        <a translation-url href="#" ng-click="gotoList($event,'contact')" class="color-darkgrey"
+                                            style="font-weight:500;">
+                                            Contacts <span class="hidden-xs">- people and organizations</span> (CON)</a>
```
Comment (blaisefonseca) [inline app/views/register/dashboard.html]: will `people and organizations` confuse users with Organization under reference section

File: app/views/register/dashboard.html [hunk 6]
```diff
@@ -167,180 +182,194 @@ <h4 style="padding-bottom:3px;margin-top:0;border-bottom:1px solid #ddd;">
...
+                    </div>
+                </div>
+
+            </div>
 
+            <div class="row" style="padding:0;margin:0 1px 0 1px; border-bottom:1px solid #ddd; ">
+                <span class="color-orange"> Reference records
+                    <span href="#" class="helpinfoInline fa fa-info-circle color-litegrey" data-toggle="tooltip"
+                        data-html="true" data-title="Reference records contain information relevant to the Protocol (but not reported as part of a Party’s obligations under the
+                            Protocol). This information can be made available to the ABS Clearing-House by any registered user
```
Comment (blaisefonseca) [inline app/views/register/dashboard.html]: can we have a generic language as this will be displayed for BCH as well?

**Comment by blaisefonseca** [comment]
can you merger top-request and top-records directive

**Comment by mattdias** [comment]
> can you merger top-request and top-records directive

No, they do different things

File: app/views/directives/home-articles-small.html (hunk 2 above)
Comment (mattdias) [reply app/views/directives/home-articles-small.html]: I want to display all or add paging. which i was saving for the  next release

File: app/views/directives/home-articles-small.js (hunk 1 above)
Comment (mattdias) [reply app/views/directives/home-articles-small.js]: i want to also display the total number of articles.

File: app/views/directives/task-id-directive.html (hunk 3 above)
Comment (mattdias) [reply app/views/directives/task-id-directive.html]: ok

Comment (mattdias) [reply app/views/directives/task-id-directive.html]: you are right but I think Admin should be able to cancel a request, no?

File: app/views/forms/view/view-articles.html (hunk 4 above)
Comment (mattdias) [reply app/views/forms/view/view-articles.html]: I want to show all articles, until we add paging

File: app/views/register/dashboard.html (hunk 5 above)
Comment (mattdias) [reply app/views/register/dashboard.html]: changed

File: app/views/directives/home-articles-small.js (hunk 1 above)
Comment (mattdias) [reply app/views/directives/home-articles-small.js]: fixed

File: app/views/directives/home-articles-small.html (hunk 2 above)
Comment (mattdias) [reply app/views/directives/home-articles-small.html]: fixed

File: app/views/directives/task-id-directive.html (hunk 3 above)
Comment (mattdias) [reply app/views/directives/task-id-directive.html]: fixed

File: app/views/forms/view/view-articles.html (hunk 4 above)
Comment (mattdias) [reply app/views/forms/view/view-articles.html]: fixed

File: app/views/register/dashboard.html (hunk 5 above)
Comment (mattdias) [reply app/views/register/dashboard.html]: fixed

File: app/views/register/dashboard.html (hunk 6 above)
Comment (mattdias) [reply app/views/register/dashboard.html]: fixed

File: app/views/directives/home-articles-small.js [hunk 7]
```diff
@@ -21,15 +21,28 @@ define(['app',
...
+                            })
+                          
+                            ag.push({"$count" : "mycount"});
+                            var qsCount = {
+                                "ag" : JSON.stringify(ag) 
+                             };
+                            
+                            articlesService.getArticles(qsCount).then(function(data){
+                                $scope.articlesCount = (data[0]||{}).mycount||0;
                               })
```
Comment (blaisefonseca) [inline app/views/directives/home-articles-small.js]: execute both queries in one call $q.all[] so you get the result in one response

File: app/views/register/requests.js [hunk 8]
```diff
@@ -73,6 +82,12 @@ define(['app',
                     load (query, 0)
 
                  };
+                 //==================================
+                 $scope.$watch('filterType', function(old, newVal){
+                    if(old != newVal)
+                        $scope.search();
+                }, true);
```
Comment (blaisefonseca) [inline app/views/register/requests.js]: use ng-change

File: app/views/register/requests.js [hunk 9]
```diff
@@ -28,6 +29,14 @@ define(['app',
                 $scope.itemCount=0;
                 $scope.itemsPerPage=15;
 
+                $scope.options = {
+                    filterTypes: function () {
+                        return _.sortBy(_.map(realm.schemas, function(schema, key){
```
Comment (blaisefonseca) [inline app/views/register/requests.js]: remove _.sortBy

File: app/views/forms/view/view-articles.html [hunk 10]
```diff
@@ -43,8 +43,8 @@ <h3 style="font-size:160%;margin-bottom:30px;">{{article.title  | lstring:locale
             </div>
         </div>
     </div>  
-
-    <div pagination ng-if="itemCount > itemsPerPage" class="text-center"  current-page="currentPage"  page-count="pageCount" on-page-change="onPageChange(page)"></div> 
+        {{itemCount}} , {{currentPage}} , {{pageCount}}, {{itemsPerPage}}
```
Comment (blaisefonseca) [inline app/views/forms/view/view-articles.html]: take out

File: app/views/register/requests.js (hunk 9 above)
Comment (mattdias) [reply app/views/register/requests.js]: done

File: app/views/forms/view/view-articles.html (hunk 10 above)
Comment (mattdias) [reply app/views/forms/view/view-articles.html]: done

File: app/views/register/requests.js (hunk 8 above)
Comment (mattdias) [reply app/views/register/requests.js]: didnt work for me.

**Pull request merged**